Lib/test/test_mimetypes.py

```diff
@@ -67,6 +67,20 @@ def test_read_mime_types(self):
             mime_dict = mimetypes.read_mime_types(file)
             eq(mime_dict[".pyunit"], "x-application/x-unittest")
 
+        # bpo-41048: read_mime_types should read the rule file with 'utf-8' encoding.
+        # Not with locale encoding. _bootlocale has been imported because io.open(...)
+        # uses it.
+        with support.temp_dir() as directory:
+            data = "application/no-mans-land  Fran\u00E7ais"
+            file = pathlib.Path(directory, "sample.mimetype")
+            file.write_text(data)
```

Comment:
`write_text()` uses the locale encoding by default. Either specify the encoding or use `write_bytes()`.